### ISSUE TITLE
[Slider] Thumb focus/focus-visible broken due to `onPointerDown` implementation?

### ISSUE THREAD

**Issue by soykje** [radix-ui/primitives#2397]
### Current Behavior
`:focus-visible` pseudo-class is meant to help target keyboard navigation focus, and improve accessibility. But for some reason on `<Slider.Thumb />` component, related styles are being applied on both `focus` and `focus-visible` states.

After some investigations it seems related to `onPointerDown` implementation with "forced" focus (see [here](https://github.com/radix-ui/primitives/blob/83a8c13bf66f3d9f17d77caeb187a69eb146930b/packages/react/slider/src/Slider.tsx#L388)). Comments weren't so clear to me, so I couldn't exactly see how to fix that trough a PR :disappointed: 

### Expected behavior
The `:focus-visible` styles should only apply on `focus-visible` state, and not on `focus` one too.

### Reproducible example
In below example you'll be able to reproduce by clicking on thumb, and see `focus-visible` styles applied.
[CodeSandbox Template](https://codesandbox.io/p/sandbox/jovial-sara-4z7nlm)

### Suggested solution
There is actually a focus event option (`focusVisible`) that could help, but it is only available on some browsers for now: https://developer.mozilla.org/en-US/docs/Web/API/HTMLElement/focus#parameters.
Or maybe there would be a way to target touch devices differently, to avoid this problem?

Thx in advance for your help!!! :pray: 

### Your environment

| Software         | Name(s) | Version |
| ---------------- | ------- | ------- |
| Radix Package(s) | react-slider | 1.1.2 |
| React            | n/a     | ^16.8 || ^17.0 || ^18.0 |
| Browser          | Brave | 1.58.127 |
| Node             | n/a     | 18.16.0|
| npm/yarn         | npm | 9.6.7 |
| Operating System | Ubuntu | 22.04.3 LTS |

**Comment by soykje** [comment]
I locally tested using a custom `onPointerDown` callback, with `target.focus({ focusVisible: false })` and it worked as expected, but on Firefox only it seems :confused: Also all the logic behind this implementation (with `setPointerCapture`) is still not clear to me, for now at least.

**Comment by benoitgrelard** [comment]
Yeah ideally, the browser would give us better low-level control. It's good to see things are moving in this area.
